In [103]:
# imports
import sys
sys.path.append('../') 
import requests
import pandas as pd
import os # use this to access your environment variables
from citybikes import get_city_bikes_info
from apicall import get_venues_fs,query_yelp
import json


# Foursquare

POI - Vegetarian, Library
| City Bike Location - Toronto

Send a request to Foursquare with a small radius (1000m) for all the bike stations in your city of choice. 

In [ ]:
FOURSQUARE_KEY = os.environ['FOURSQUARE_KEY']
foursquare_api_key = "FOURSQUARE_KEY"
yelp_api_key = "YELP_KEY"
network_id="bixi-toronto"
df_four=pd.DataFrame()
df_yelp=pd.DataFrame()

#call citybikes to fecth location details for all sations in the selected city 
# I have ran this api call to fetch citybikes station , lat, log and saved it to a csv file. 
# fetching info for this from csv file

file_path = '/Users/priyaganesan/Desktop/python_exercise/Pandas_exercise/db/Final-Project-statistical-Modelling-with-Python/bike_info_csv.csv'

# Read the CSV file into a DataFrame
bike_info_df = pd.read_csv(file_path)

point_of_interest_coordinates = bike_info_df[['latitude', 'longitude','name']]
#point_of_interest_coordinates=point_of_interest_coordinates.head(5)
# for each station lat,long call foursquare 
for index, row in point_of_interest_coordinates.iterrows():
    latitude = row['latitude']
    longitude = row['longitude']
    station_name=row['name']
    #Call function to fetch points of interest near each lat,long from foursquare ,12080-library
    #,13377-vegetarian friendly
    #CALL FOURSQUARE API
    f_res=get_venues_fs(latitude,longitude,1000,four_square_key,'13377,12080') 
    #copy foursquare json data into df 
    fdf=pd.DataFrame(["category","name","tel","tips","distance","latitude","longitude","hours","address","station_name"])
    extracted_data_four=[
        {
            'name': results.get('name'),
            'distance':results.get('distance'),
            'category': results.get('categories', [])[0].get('short_name'),
            'latitude': results.get('geocodes', {}).get('main', {}).get('latitude'),
            'longitude': results.get('geocodes', {}).get('main', {}).get('longitude'),
            'rating': results.get('rating'),
            'reviews':results.get('stats',{}).get('total_ratings'),
            'description':results.get('description'),
            'formatted_address': results.get('location', {}).get('formatted_address'),
            'telephone': results.get('tel'),
            'tips': results.get('tips', [])[0].get('text') if results.get('tips') else None,
            'hours': results.get('hours', {}).get('display'),
            'station_name': station_name
        }
        for results in f_res.get('results', [])
    ]
    #COPY PARSED FLATTENED DATA TO DATAFRAME
    df = pd.DataFrame(extracted_data_four)
    df_four=df_four._append(df,ignore_index=True)

    #CALL YELP API 
    y_res=query_yelp(latitude,longitude,yelp_api_key) 
    #print(y_res)
    #copy json yelp data into df 
    ydy=pd.DataFrame(["name","closed","review","rating","categories","phone","address","distance","latitude","longitude","station_name"])
    for business in y_res.get('businesses', []):
    # Extract display address as a string
        display_address = ', '.join(business.get('location', {}).get('display_address', []))
    
    extracted_data_yelp=[
        {
            'name': business.get('name'),
            'closed':business.get('is_closed'),
            'review':business.get('review_count'),
            'rating':business.get('rating'),
            'categories': business.get('categories', [])[0].get('title'),
            'phone':business.get('phone'),
            'distance':business.get('distance'),
            'latitude': business.get('coordinates', {}).get('latitude'),
            'longitude': business.get('coordinates', {}).get('longitude'),
            'address':display_address,
            #'address':business.get('location', {}).get('display_address'),
            'station_name': station_name
        }
        for business in y_res.get('businesses', [])
    ]
    #COPY PARSED FLATTENED DATA TO DATAFRAME
    df = pd.DataFrame(extracted_data_yelp)
    df_yelp=df_yelp._append(df,ignore_index=True)  
#copy df into csv for further analysis
df_four.to_csv('/Users/priyaganesan/Desktop/python_exercise/Pandas_exercise/db/Final-Project-statistical-Modelling-with-Python/foursquare.csv', index=False)  
df_yelp.to_csv('/Users/priyaganesan/Desktop/python_exercise/Pandas_exercise/db/Final-Project-statistical-Modelling-with-Python/yelp.csv', index=False)  

  



In [193]:
bike_info_df.head(10)

0                                       name  free_bikes   latitude  \
0 NaN                  Queen St E / Woodward Ave         7.0  43.665269   
1 NaN                Primrose Ave / Davenport Rd         3.0  43.671420   
2 NaN                    Queen St E / Rhodes Ave        20.0  43.666224   
3 NaN                       Bond St / Queen St E        31.0  43.653236   
4 NaN                   Church St / Alexander St        14.0  43.663722   
5 NaN  University Ave / Gerrard St W (East Side)        18.0  43.657763   
6 NaN                       Edward St / Yonge St        23.0  43.656729   
7 NaN                    Euclid Ave / Bloor St W         0.0  43.664467   
8 NaN                      Yonge St / Aylmer Ave        11.0  43.675492   
9 NaN                   Davenport Rd / Avenue Rd         3.0  43.674991   

   longitude  
0 -79.319796  
1 -79.445947  
2 -79.317693  
3 -79.376716  
4 -79.380288  
5 -79.389165  
6 -79.382736  
7 -79.414783  
8 -79.388858  
9 -79.396273

In [195]:
df_yelp.head()

name  closed  review  rating  categories         phone  \
0             Udupi Palace   False     208     3.5      Indian  +14164058189   
1             KaSpace Cafe   False       9     5.0  Vegetarian  +16476579600   
2         Bombay Chowpatty   False      59     3.0      Indian  +14164058080   
3      The Social Gardener   False       5     3.5       Vegan  +14164656021   
4  Gerrard/Ashdale Library   False       4     3.5   Libraries  +14163937717   

      distance   latitude  longitude  \
0   782.410965  43.672045 -79.322420   
1  1023.018862  43.662630 -79.331900   
2   832.357744  43.671982 -79.324374   
3   839.372147  43.671644 -79.325385   
4   802.887228  43.672482 -79.322082   

                                             address  \
0  1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   
1  1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   
2  1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   
3  1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   
4  1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   

                station_name  
0  Queen St E / Woodward Ave  
1  Queen St E / Woodward Ave  
2  Queen St E / Woodward Ave  
3  Queen St E / Woodward Ave  
4  Queen St E / Woodward Ave

Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

In [219]:
df_four.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4216 entries, 0 to 4215
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   name               4216 non-null   object 
 1   distance           4216 non-null   int64  
 2   category           4216 non-null   object 
 3   latitude           4216 non-null   float64
 4   longitude          4216 non-null   float64
 5   rating             2104 non-null   float64
 6   reviews            2104 non-null   float64
 7   description        583 non-null    object 
 8   formatted_address  4216 non-null   object 
 9   telephone          4102 non-null   object 
 10  tips               2812 non-null   object 
 11  hours              3649 non-null   object 
 12  station_name       4216 non-null   object 
dtypes: float64(4), int64(1), object(8)
memory usage: 428.3+ KB


In [198]:
df_yelp.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3677 entries, 0 to 3676
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   name          3677 non-null   object 
 1   closed        3677 non-null   bool   
 2   review        3677 non-null   int64  
 3   rating        3677 non-null   float64
 4   categories    3677 non-null   object 
 5   phone         3677 non-null   object 
 6   distance      3677 non-null   float64
 7   latitude      3677 non-null   float64
 8   longitude     3677 non-null   float64
 9   address       3677 non-null   object 
 10  station_name  3677 non-null   object 
dtypes: bool(1), float64(4), int64(1), object(5)
memory usage: 291.0+ KB


In [199]:
df_yelp.head()

name  closed  review  rating  categories         phone  \
0             Udupi Palace   False     208     3.5      Indian  +14164058189   
1             KaSpace Cafe   False       9     5.0  Vegetarian  +16476579600   
2         Bombay Chowpatty   False      59     3.0      Indian  +14164058080   
3      The Social Gardener   False       5     3.5       Vegan  +14164656021   
4  Gerrard/Ashdale Library   False       4     3.5   Libraries  +14163937717   

      distance   latitude  longitude  \
0   782.410965  43.672045 -79.322420   
1  1023.018862  43.662630 -79.331900   
2   832.357744  43.671982 -79.324374   
3   839.372147  43.671644 -79.325385   
4   802.887228  43.672482 -79.322082   

                                             address  \
0  1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   
1  1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   
2  1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   
3  1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   
4  1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   

                station_name  
0  Queen St E / Woodward Ave  
1  Queen St E / Woodward Ave  
2  Queen St E / Woodward Ave  
3  Queen St E / Woodward Ave  
4  Queen St E / Woodward Ave

In [200]:
df_yelp.tail()

name  closed  review  \
3672                                            Freshii   False       6   
3673                           City Hall Public Library   False       6   
3674  Toronto Public Library - Parliament Street Branch   False       6   
3675                 Toronto Public Library - Riverdale   False       4   
3676     Toronto Public Library - St. James Town Branch   False       3   

      rating categories         phone     distance   latitude  longitude  \
3672     2.0      Salad  +16474848550   790.938858  43.651226 -79.364586   
3673     3.5  Libraries  +14163937650   776.740866  43.653347 -79.383826   
3674     2.0  Libraries  +14163937663   314.733258  43.661830 -79.367910   
3675     4.0  Libraries  +14163937720  1010.279295  43.665810 -79.353140   
3676     4.0  Libraries  +14163937744  1068.656684  43.667890 -79.374690   

                                                address  \
3672  100 Queen Street W, Nathan Phillips Square, To...   
3673  100 Queen Street W, Nathan Phillips Square, To...   
3674     495 Sherbourne St, Toronto, ON M4X 1L1, Canada   
3675     495 Sherbourne St, Toronto, ON M4X 1L1, Canada   
3676     495 Sherbourne St, Toronto, ON M4X 1L1, Canada   

                           station_name  
3672  Richmond St E / Jarvis St Green P  
3673  Richmond St E / Jarvis St Green P  
3674              Oak St / Sackville St  
3675              Oak St / Sackville St  
3676              Oak St / Sackville St

# Comparing Results

Which API provided you with more complete data? Provide an explanation. 

FOURSQUARE

Get the top 10 restaurants according to their rating

Top 10 POI- Vegetarian restaurant by rating

Remove duplicates and consider POI NAME, RATING, REVIEW, ADDRESS, TELEPHONE TO DISPLAY TOP 10 RATED RESTAURANT

In [228]:
columns_to_copy=['name','category','rating','reviews','formatted_address','telephone']
df_four_selectedcolumn=df_four[columns_to_copy].copy()
df_four_selectedcolumn=df_four_selectedcolumn.drop_duplicates()


In [241]:
foursquare_top_poi=df_four_selectedcolumn.sort_values(by='rating',ascending=False)
foursquare_top_resturants=foursquare_top_poi[foursquare_top_poi['category']!='Library'].head(20)
foursquare_top_resturants


name                         category  rating  reviews  \
368         Fresh  Vegan and Vegetarian Restaurant     7.8    140.0   
31   Saigon Lotus  Vegan and Vegetarian Restaurant     7.8     18.0   
347           Gia  Vegan and Vegetarian Restaurant     7.8      9.0   
237   King's Cafe                             Café     7.7     67.0   
324    Grey Tiger                         Cocktail     7.7     16.0   

                                     formatted_address       telephone  
368  90 Eglinton Ave E (Yonge & Eglinton), Toronto ...  (416) 599-4442  
31    6 St Andrew St (Spadina Ave), Toronto ON M5T 1K6  (416) 901-1115  
347               1214 Dundas St W, Toronto ON M6J 1X5  (416) 535-8888  
237                192 Augusta Ave, Toronto ON M5T 2L6  (416) 591-1340  
324                1190 Bloor St W, Toronto ON M6H 1N2  (647) 925-3310

In [238]:
foursquare_top_libraries=foursquare_top_poi[foursquare_top_poi['category']=='Library'].head(20)
foursquare_top_libraries

name category  rating  \
59    Toronto Public Library - Toronto Reference Lib...  Library     9.0   
189                              Toronto Public Library  Library     8.6   
30                               Toronto Public Library  Library     8.3   
89                               Toronto Public Library  Library     8.0   
325                              Toronto Public Library  Library     7.9   
1597                             Toronto Public Library  Library     7.8   
146                              Toronto Public Library  Library     7.6   
3687                             Toronto Public Library  Library     7.6   
443                              Toronto Public Library  Library     7.5   
183                              Toronto Public Library  Library     7.5   
369                              Toronto Public Library  Library     7.5   
2246                                York Public Library  Library     7.4   
3263                           Brentwood Public Library  Library     7.4   
515                              Toronto Public Library  Library     7.1   
15                               Toronto Public Library  Library     6.9   
261                              Toronto Public Library  Library     6.8   
2230                         Long Branch Public Library  Library     6.5   
20                               Toronto Public Library  Library     6.5   
180                              Toronto Public Library  Library     6.3   
54           Toronto Public Library - Palmerston Branch  Library     6.1   

      reviews                                  formatted_address  \
59      378.0  789 Yonge St (btwn Collier & Asquith Ave), Tor...   
189      42.0             190 Fort York Blvd, Toronto ON M5V 0E7   
30       38.0  239 College St (at Huron St.), Toronto ON M5T 1R5   
89       11.0  370 Broadview Ave (at Gerrard St E), Toronto O...   
325      51.0  1101 Bloor St W (at Gladstone), Toronto ON M6H...   
1597     15.0  3083 Yonge St (at Lawrence Ave.), Toronto ON M...   
146      14.0  701 Pape Ave (at Danforth Ave), Toronto ON M4K...   
3687      8.0              496 Birchmount Rd, Toronto ON M1K 1N8   
443      15.0   2161 Queen St E (at Lee St.), Toronto ON M4L 1J1   
183      12.0  40 St Clair Ave E (at Yonge St.), Toronto ON M...   
369      40.0  40 Orch View Blvd (at Yonge St.), Toronto ON M...   
2246     11.0   620 Jane St (at Dundas St W), Toronto ON M6S 4A6   
3263      9.0  36 Brentwood Rd N (at Bloor St. W), Toronto ON...   
515      11.0  1303 Queen St W (at Cowan Ave.), Toronto ON M6...   
15       15.0     100 Queen St W (City Hall), Toronto ON M5H 2N3   
261       9.0  327 Bathurst St (at Dundas St.), Toronto ON M5...   
2230     10.0         3500 Lake Shore Blvd W, Toronto ON M8W 1N6   
20       12.0  495 Sherbourne St (Wellesley Ave), Toronto ON ...   
180      10.0  1431 Bathurst St (at Melgund Rd.), Toronto ON ...   
54        8.0  560 Palmerston Ave (at Bloor St. W), Toronto O...   

           telephone  
59    (416) 395-5577  
189   (416) 393-6240  
30    (416) 393-7746  
89    (416) 393-7720  
325   (416) 393-7674  
1597  (416) 393-7730  
146   (416) 393-7727  
3687  (416) 396-8890  
443   (416) 393-7703  
183   (416) 393-7657  
369   (416) 393-7610  
2246  (416) 394-1014  
3263  (416) 394-5240  
515   (416) 393-7686  
15    (416) 393-7650  
261   (416) 393-7653  
2230  (416) 394-5320  
20    (416) 393-7744  
180   (416) 393-7683  
54    (416) 393-7680

YELP

Top 10 POI- Vegetarian restaurants by rating

In [230]:
df_yelp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3677 entries, 0 to 3676
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   name          3677 non-null   object 
 1   closed        3677 non-null   bool   
 2   review        3677 non-null   int64  
 3   rating        3677 non-null   float64
 4   categories    3677 non-null   object 
 5   phone         3677 non-null   object 
 6   distance      3677 non-null   float64
 7   latitude      3677 non-null   float64
 8   longitude     3677 non-null   float64
 9   address       3677 non-null   object 
 10  station_name  3677 non-null   object 
dtypes: bool(1), float64(4), int64(1), object(5)
memory usage: 291.0+ KB


YELP- Remove duplicates and consider POI NAME, RATING, REVIEW, ADDRESS, TELEPHONE TO DISPLAY TOP 10 RATED RESTAURANT

In [231]:

columns_to_copy=['name','categories','rating','review','address','phone']
df_yelp_selectedcolumn=df_yelp[columns_to_copy].copy()
df_yelp_selectedcolumn=df_yelp_selectedcolumn.drop_duplicates()


In [239]:
yelp_top_poi=df_yelp_selectedcolumn.sort_values(by='rating',ascending=False)
yelp_top_resturants=yelp_top_poi[(yelp_top_poi['categories']!='Libraries') & (yelp_top_poi['categories']!='Museums')]
yelp_top_resturants.head(20)


name  categories  rating  review  \
277       Sunny Cafe  Vegetarian     5.0       3   
1908    KaSpace Cafe  Vegetarian     5.0       9   
3350    KaSpace Cafe  Vegetarian     5.0       9   
1931      Sunny Cafe  Vegetarian     5.0       3   
1       KaSpace Cafe  Vegetarian     5.0       9   
127       Sunny Cafe  Vegetarian     5.0       3   
98        Sunny Cafe  Vegetarian     5.0       3   
198     KaSpace Cafe  Vegetarian     5.0       9   
3031      Sunny Cafe  Vegetarian     5.0       3   
971       Sunny Cafe  Vegetarian     5.0       3   
664       Sunny Cafe  Vegetarian     5.0       3   
578       Sunny Cafe  Vegetarian     5.0       3   
1992      Sunny Cafe  Vegetarian     5.0       3   
2813    Saigon Lotus  Vietnamese     4.5     221   
889     Saigon Lotus  Vietnamese     4.5     221   
2814  Buddha's Vegan  Vegetarian     4.5     148   
897    Kupfert & Kim  Vegetarian     4.5       6   
959       Mad Radish       Salad     4.5       9   
2770  Buddha's Vegan  Vegetarian     4.5     148   
292    Kupfert & Kim  Vegetarian     4.5       6   

                                                address         phone  
277   40 Saint George Street, Room 6141, Toronto, ON...                
1908    765 Queen Street E, Toronto, ON M4M 1H3, Canada  +16476579600  
3350  1326 Gerrard Street  E, Toronto, ON M4L 1Z1, C...  +16476579600  
1931      789 Yonge Street, Toronto, ON M4W 2G8, Canada                
1     1432 Gerrard Street East, Toronto, ON M4L 1Z6,...  +16476579600  
127       2 Bloor Street W, Toronto, ON M4W 3E2, Canada                
98         10 Spadina Road, Toronto, ON M5R 2S7, Canada                
198    Jones 118 Jones Ave, Toronto, ON M4M 2Z9, Canada  +16476579600  
3031          55 Avenue Rd, Toronto, ON M5R 3L2, Canada                
971     322 Bloor Street W, Toronto, ON M5S 1W5, Canada                
664   15 King's College Cir, 2nd Fl, N Wing, Toronto...                
578   10 King's College Road, Room 2402, Toronto, ON...                
1992  113 St. Joseph Street, Toronto, ON M5S 3C2, Ca...                
2813    1 Devonshire Place, Toronto, ON M5S 1H8, Canada  +14169011115  
889    327 Bathurst Street, Toronto, ON M5T 1J1, Canada  +14169011115  
2814    1 Devonshire Place, Toronto, ON M5S 1H8, Canada  +14166033811  
897    327 Bathurst Street, Toronto, ON M5T 1J1, Canada  +14162607777  
959     322 Bloor Street W, Toronto, ON M5S 1W5, Canada  +14167920700  
2770  40 Saint George Street, Room 6141, Toronto, ON...  +14166033811  
292          20 Bay Street, Toronto, ON M5J 2N8, Canada  +14162607777

In [240]:
yelp_top_libraries=yelp_top_poi[(yelp_top_poi['categories']=='Libraries') & (yelp_top_poi['categories']!='Museums')]
yelp_top_libraries.head(20)


name categories  rating  review  \
1928  John W. Graham Library  Libraries     5.0       2   
375     John M Kelly Library  Libraries     5.0       1   
276   John W. Graham Library  Libraries     5.0       2   
1889  John W. Graham Library  Libraries     5.0       2   
65    John W. Graham Library  Libraries     5.0       2   
1243    John M Kelly Library  Libraries     5.0       1   
1249  John W. Graham Library  Libraries     5.0       2   
1700  John W. Graham Library  Libraries     5.0       2   
145   John W. Graham Library  Libraries     5.0       2   
3119  John W. Graham Library  Libraries     5.0       2   
2832  John W. Graham Library  Libraries     5.0       2   
2380    John M Kelly Library  Libraries     5.0       1   
2666    John M Kelly Library  Libraries     5.0       1   
1556  John W. Graham Library  Libraries     5.0       2   
2687    John M Kelly Library  Libraries     5.0       1   
329   John W. Graham Library  Libraries     5.0       2   
2679  John W. Graham Library  Libraries     5.0       2   
1827  John W. Graham Library  Libraries     5.0       2   
1067    John M Kelly Library  Libraries     5.0       1   
1129  John W. Graham Library  Libraries     5.0       2   

                                                address         phone  
1928      789 Yonge Street, Toronto, ON M4W 2G8, Canada  +14169785851  
375   113 St. Joseph Street, Toronto, ON M5S 3C2, Ca...                
276   40 Saint George Street, Room 6141, Toronto, ON...  +14169785851  
1889  Toronto Reference Library, 789 Yonge St, 2nd f...  +14169785851  
65    10 King's College Road, Room 2402, Toronto, ON...  +14169785851  
1243      341 Yonge Street, Toronto, ON M5B 1S1, Canada                
1249      341 Yonge Street, Toronto, ON M5B 1S1, Canada  +14169785851  
1700  118-92 King Street E, Toronto, ON M5C 2V8, Canada  +14169785851  
145   15 King's College Cir, 2nd Fl, N Wing, Toronto...  +14169785851  
3119     56 Wellesley St W, Toronto, ON M5S 2S3, Canada  +14169785851  
2832    1 Devonshire Place, Toronto, ON M5S 1H8, Canada  +14169785851  
2380      372 Yonge Street, Toronto, ON M5G 2K9, Canada                
2666  120 St. George Street, Toronto, ON M5S 3H1, Ca...                
1556  113 St. Joseph Street, Toronto, ON M5S 3C2, Ca...  +14169785851  
2687         65 Queen St W, Toronto, ON M5H 2M5, Canada                
329   292 Brunswick Avenue, 2nd Floor, Toronto, ON M...  +14169785851  
2679         65 Queen St W, Toronto, ON M5H 2M5, Canada  +14169785851  
1827   327 Bathurst Street, Toronto, ON M5T 1J1, Canada  +14169785851  
1067      2 Bloor Street W, Toronto, ON M4W 3E2, Canada                
1129    252 Bloor Street W, Toronto, ON M5S 1V6, Canada  +14169785851

In [203]:
print(df_yelp.dtypes)

name             object
closed             bool
review            int64
rating          float64
categories       object
phone            object
distance        float64
latitude        float64
longitude       float64
address          object
station_name     object
dtype: object


In [224]:
foursquare_top_poi=df_four.sort_values(by='rating',ascending=False)
foursquare_top_resturants=foursquare_top_poi[foursquare_top_poi['category']!='Library'].head(10)
foursquare_top_resturants

name  distance  \
1338  Fresh Kitchen + Juice Bar Front       497   
1380  Fresh Kitchen + Juice Bar Front       688   
1349  Fresh Kitchen + Juice Bar Front       498   
210   Fresh Kitchen + Juice Bar Front       701   
1390  Fresh Kitchen + Juice Bar Front       402   
2980  Fresh Kitchen + Juice Bar Front       535   
2940  Fresh Kitchen + Juice Bar Front       908   
2885  Fresh Kitchen + Juice Bar Front       661   
1328  Fresh Kitchen + Juice Bar Front        85   
3591  Fresh Kitchen + Juice Bar Front       898   

                             category   latitude  longitude  rating  reviews  \
1338  Vegan and Vegetarian Restaurant  43.647823 -79.374336     8.9     19.0   
1380  Vegan and Vegetarian Restaurant  43.647823 -79.374336     8.9     19.0   
1349  Vegan and Vegetarian Restaurant  43.647823 -79.374336     8.9     19.0   
210   Vegan and Vegetarian Restaurant  43.647823 -79.374336     8.9     19.0   
1390  Vegan and Vegetarian Restaurant  43.647823 -79.374336     8.9     19.0   
2980  Vegan and Vegetarian Restaurant  43.647823 -79.374336     8.9     19.0   
2940  Vegan and Vegetarian Restaurant  43.647823 -79.374336     8.9     19.0   
2885  Vegan and Vegetarian Restaurant  43.647823 -79.374336     8.9     19.0   
1328  Vegan and Vegetarian Restaurant  43.647823 -79.374336     8.9     19.0   
3591  Vegan and Vegetarian Restaurant  43.647823 -79.374336     8.9     19.0   

     description                                  formatted_address  \
1338        None  47 Front St E (Church Street), Toronto ON M5E 1B3   
1380        None  47 Front St E (Church Street), Toronto ON M5E 1B3   
1349        None  47 Front St E (Church Street), Toronto ON M5E 1B3   
210         None  47 Front St E (Church Street), Toronto ON M5E 1B3   
1390        None  47 Front St E (Church Street), Toronto ON M5E 1B3   
2980        None  47 Front St E (Church Street), Toronto ON M5E 1B3   
2940        None  47 Front St E (Church Street), Toronto ON M5E 1B3   
2885        None  47 Front St E (Church Street), Toronto ON M5E 1B3   
1328        None  47 Front St E (Church Street), Toronto ON M5E 1B3   
3591        None  47 Front St E (Church Street), Toronto ON M5E 1B3   

           telephone                                    tips  \
1338  (647) 693-7556  Buffalo cauliflower. Got tons of zing!   
1380  (647) 693-7556  Buffalo cauliflower. Got tons of zing!   
1349  (647) 693-7556  Buffalo cauliflower. Got tons of zing!   
210   (647) 693-7556  Buffalo cauliflower. Got tons of zing!   
1390  (647) 693-7556  Buffalo cauliflower. Got tons of zing!   
2980  (647) 693-7556  Buffalo cauliflower. Got tons of zing!   
2940  (647) 693-7556  Buffalo cauliflower. Got tons of zing!   
2885  (647) 693-7556  Buffalo cauliflower. Got tons of zing!   
1328  (647) 693-7556  Buffalo cauliflower. Got tons of zing!   
3591  (647) 693-7556  Buffalo cauliflower. Got tons of zing!   

                                         hours  \
1338  Mon-Fri 11:00-22:00; Sat-Sun 10:30-22:00   
1380  Mon-Fri 11:00-22:00; Sat-Sun 10:30-22:00   
1349  Mon-Fri 11:00-22:00; Sat-Sun 10:30-22:00   
210   Mon-Fri 11:00-22:00; Sat-Sun 10:30-22:00   
1390  Mon-Fri 11:00-22:00; Sat-Sun 10:30-22:00   
2980  Mon-Fri 11:00-22:00; Sat-Sun 10:30-22:00   
2940  Mon-Fri 11:00-22:00; Sat-Sun 10:30-22:00   
2885  Mon-Fri 11:00-22:00; Sat-Sun 10:30-22:00   
1328  Mon-Fri 11:00-22:00; Sat-Sun 10:30-22:00   
3591  Mon-Fri 11:00-22:00; Sat-Sun 10:30-22:00   

                               station_name  
1338           Frederick St / The Esplanade  
1380                 Queens Quay / Yonge St  
1349         King St W / Bay St (West Side)  
210                     King St W / York St  
1390       Front St W / Bay St (North Side)  
2980      Richmond St E / Jarvis St Green P  
2940  Mutual St / Shuter St (Arena Gardens)  
2885             Bay St / Harbour St (East)  
1328                  Front St E / Scott St  
3591       York St / Harbour St (Love Park)

In [205]:
yelp_top_poi=df_yelp.sort_values(by='rating',ascending=False)
yelp_top_resturants=yelp_top_poi[(yelp_top_poi['categories']!='Library') & (yelp_top_poi['categories']!='Museums')]
yelp_top_resturants.head(10)

name  closed  review  rating  categories  \
2438  John W. Graham Library   False       2     5.0   Libraries   
198             KaSpace Cafe   False       9     5.0  Vegetarian   
1309              Sunny Cafe   False       3     5.0  Vegetarian   
1               KaSpace Cafe   False       9     5.0  Vegetarian   
753             KaSpace Cafe   False       9     5.0  Vegetarian   
3522  John W. Graham Library   False       2     5.0   Libraries   
3521              Sunny Cafe   False       3     5.0  Vegetarian   
741               Sunny Cafe   False       3     5.0  Vegetarian   
737   John W. Graham Library   False       2     5.0   Libraries   
726               Sunny Cafe   False       3     5.0  Vegetarian   

             phone     distance  latitude  longitude  \
2438  +14169785851   828.899614  43.66501  -79.39660   
198   +16476579600   835.200890  43.66263  -79.33190   
1309                 673.595590  43.66694  -79.40321   
1     +16476579600  1023.018862  43.66263  -79.33190   
753   +16476579600   454.047204  43.66263  -79.33190   
3522  +14169785851   520.594666  43.66501  -79.39660   
3521                 325.968568  43.66694  -79.40321   
741                  648.193619  43.66694  -79.40321   
737   +14169785851  1210.333758  43.66501  -79.39660   
726                 1240.193859  43.66694  -79.40321   

                                                address  \
2438  10 King's College Road, Room 2402, Toronto, ON...   
198    Jones 118 Jones Ave, Toronto, ON M4M 2Z9, Canada   
1309       10 Spadina Road, Toronto, ON M5R 2S7, Canada   
1     1432 Gerrard Street East, Toronto, ON M4L 1Z6,...   
753    Jones 118 Jones Ave, Toronto, ON M4M 2Z9, Canada   
3522  40 Saint George Street, Room 6141, Toronto, ON...   
3521  40 Saint George Street, Room 6141, Toronto, ON...   
741   15 King's College Cir, 2nd Fl, N Wing, Toronto...   
737   15 King's College Cir, 2nd Fl, N Wing, Toronto...   
726        10 Spadina Road, Toronto, ON M5R 2S7, Canada   

                          station_name  
2438            Dalton Rd / Bloor St W  
198             Morse St / Eastern Ave  
1309           Bathurst Subway Station  
1            Queen St E / Woodward Ave  
753      Lake Shore Blvd E / Leslie St  
3522  Spadina Ave / Sussex Ave - SMART  
3521  Spadina Ave / Sussex Ave - SMART  
741           Kendal Ave / Bernard Ave  
737           Kendal Ave / Bernard Ave  
726         Bridgman Ave / Bathurst St

In [206]:
unique_categories = df_four['category'].unique()
print(unique_categories)

['Library' 'Coffee Shop' 'Deli' 'Vegan and Vegetarian Restaurant'
 'Mexican' 'Fast Food' 'Salad' 'Burgers' 'Ice Cream' 'Caribbean'
 'Hair Salon' 'Falafel' 'Diner' 'Indian' 'Café' 'Bakery' 'Sushi' 'Buffet'
 'Hawaiian' 'Publisher' 'Thai' 'Spa' 'Lounge' 'Arts & Entertainment'
 'Chinese' 'Cocktail' 'Soup Spot' 'South Indian' 'Sandwich Spot'
 'Art Museum' 'Ethiopian' 'Architecture Firm' 'Juice Bar' 'Restaurant'
 'Gluten-free']


In [207]:
unique_categories = df_yelp['categories'].unique()
print(unique_categories)

['Indian' 'Vegetarian' 'Vegan' 'Libraries' 'Tapas/Small Plates' 'Salad'
 'Gluten-Free' 'Breakfast & Brunch' 'Museums' 'Vietnamese' 'Falafel'
 'Middle Eastern' 'Thai' 'Fast Food' 'Delis']


In [208]:
df_yelp_resturants=df_yelp[df_yelp['categories']=='Libraries']
df_yelp_resturants.tail(10)

name  closed  review  \
3643                                  Fort York Library   False      19   
3647                       Queen/Saulter Public Library   False       6   
3648                 Toronto Public Library - Riverdale   False       4   
3652         Toronto Public Library - Palmerston Branch   False       5   
3653                             Toronto Public Library   False       3   
3666         Ryerson University Student Learning Centre   False       4   
3673                           City Hall Public Library   False       6   
3674  Toronto Public Library - Parliament Street Branch   False       6   
3675                 Toronto Public Library - Riverdale   False       4   
3676     Toronto Public Library - St. James Town Branch   False       3   

      rating categories         phone     distance   latitude  longitude  \
3643     3.5  Libraries  +14163936240   802.872055  43.639007 -79.400075   
3647     3.5  Libraries  +14163937723   195.800619  43.659190 -79.347540   
3648     4.0  Libraries  +14163937720   762.709499  43.665810 -79.353140   
3652     4.0  Libraries  +14163937680   924.996991  43.665090 -79.414150   
3653     2.5  Libraries  +14163937732   664.365857  43.674570 -79.428380   
3666     3.5  Libraries  +14169795000   788.934959  43.657670 -79.381100   
3673     3.5  Libraries  +14163937650   776.740866  43.653347 -79.383826   
3674     2.0  Libraries  +14163937663   314.733258  43.661830 -79.367910   
3675     4.0  Libraries  +14163937720  1010.279295  43.665810 -79.353140   
3676     4.0  Libraries  +14163937744  1068.656684  43.667890 -79.374690   

                                                address  \
3643  1171 Dundas Street W, Toronto, ON M6J 1X3, Canada   
3647  370 Broadview Avenue, Toronto, ON M4M 2H1, Canada   
3648  370 Broadview Avenue, Toronto, ON M4M 2H1, Canada   
3652      1246 Shaw Street, Toronto, ON M6G 3N9, Canada   
3653      1246 Shaw Street, Toronto, ON M6G 3N9, Canada   
3666  100 Queen Street W, Nathan Phillips Square, To...   
3673  100 Queen Street W, Nathan Phillips Square, To...   
3674     495 Sherbourne St, Toronto, ON M4X 1L1, Canada   
3675     495 Sherbourne St, Toronto, ON M4X 1L1, Canada   
3676     495 Sherbourne St, Toronto, ON M4X 1L1, Canada   

                           station_name  
3643      Wellington St W / Stafford St  
3647                  111 Broadview Ave  
3648                  111 Broadview Ave  
3652            Christie St / Dupont St  
3653            Christie St / Dupont St  
3666  Richmond St E / Jarvis St Green P  
3673  Richmond St E / Jarvis St Green P  
3674              Oak St / Sackville St  
3675              Oak St / Sackville St  
3676              Oak St / Sackville St

In [209]:
df_four_lib=df_four[df_four['category']=='Library']
df_four_lib.tail(10)

name  distance category  \
4183                             Toronto Public Library       917  Library   
4184                          Boneshaker Reading Series       917  Library   
4185  Toronto Public Library - Toronto Reference Lib...       319  Library   
4190                             Toronto Public Library       845  Library   
4192                  Canadian Lesbian and Gay Archives       178  Library   
4195                           Brentwood Public Library       914  Library   
4202         Toronto Public Library - Palmerston Branch       203  Library   
4203                             Toronto Public Library       186  Library   
4212         Toronto Public Library - Palmerston Branch       341  Library   
4213                             Toronto Public Library       324  Library   

       latitude  longitude  rating  reviews  \
4183  43.674262 -79.460723     NaN      NaN   
4184  43.674262 -79.460723     NaN      NaN   
4185  43.671847 -79.386762     9.0    378.0   
4190  43.667774 -79.374796     6.5     12.0   
4192  43.668057 -79.384054     NaN      NaN   
4195  43.647480 -79.514244     7.4      9.0   
4202  43.665201 -79.414090     6.1      8.0   
4203  43.664934 -79.413907     NaN      NaN   
4212  43.665201 -79.414090     6.1      8.0   
4213  43.664934 -79.413907     NaN      NaN   

                             description  \
4183                                None   
4184                                None   
4185  Main intersection: Bloor and Yonge   
4190                                None   
4192                                None   
4195                                None   
4202                                None   
4203                                None   
4212                                None   
4213                                None   

                                      formatted_address       telephone  \
4183            1748 St Clair Ave W, Toronto ON M6N 1J3  (416) 393-7709   
4184                    1748 St Clair Ave W, Toronto ON  (416) 393-7710   
4185  789 Yonge St (btwn Collier & Asquith Ave), Tor...  (416) 395-5577   
4190  495 Sherbourne St (Wellesley Ave), Toronto ON ...  (416) 393-7744   
4192  34 Isabella St (at Yonge St.), Toronto ON M4Y 1N1  (416) 777-2755   
4195  36 Brentwood Rd N (at Bloor St. W), Toronto ON...  (416) 394-5240   
4202  560 Palmerston Ave (at Bloor St. W), Toronto O...  (416) 393-7680   
4203            2560 Palmerston Ave, Toronto ON M6G 2P7  (416) 393-7680   
4212  560 Palmerston Ave (at Bloor St. W), Toronto O...  (416) 393-7680   
4213            2560 Palmerston Ave, Toronto ON M6G 2P7  (416) 393-7680   

                                                   tips  \
4183                                               None   
4184                                               None   
4185  This is Toronto Public Library’s main location...   
4190  very limited number of free museum tickets. ot...   
4192  Did you know that the CLGA was founded in 1973...   
4195  Second floor is a circus of noise performed by...   
4202                              Not enough computers.   
4203                                               None   
4212                              Not enough computers.   
4213                                               None   

                                                  hours  \
4183  Tue 12:30-20:30; Wed 10:00-18:00; Thu 12:30-20...   
4184                                               None   
4185  Mon-Fri 9:00-20:30; Sat 9:00-17:00; Sun 13:30-...   
4190  Tue 12:30-20:30; Wed 10:00-18:00; Thu 12:30-20...   
4192               Tue-Thu 19:30-22:00; Fri 13:00-17:00   
4195  Mon-Fri 9:00-20:30; Sat 9:00-17:00; Sun 13:30-...   
4202  Mon 10:00-20:30; Tue 12:30-20:30; Wed 10:00-18...   
4203  Mon 10:00-20:30; Tue 12:30-20:30; Wed 10:00-18...   
4212  Mon 10:00-20:30; Tue 12:30-20:30; Wed 10:00-18...   
4213  Mon 10:00-20:30; Tue 12:30-20:30; Wed 10:00-18...   

                         station_name  
4183  Gunns Rd / Lavend

YELP - DISPLAY UNIQUE POI NAME, ADDRESS,CATEGORIES 

YELP Library

In [214]:
distinct_data = df_yelp[['name', 'address','categories']].drop_duplicates()


cnt_lib=distinct_data[distinct_data['categories']=='Libraries'].shape[0]
cnt_vegrest=distinct_data[distinct_data['categories']!='Libraries'].shape[0]
print(cnt_lib)
print(cnt_vegrest)
# # # Convert tuples back to lists if needed
# distinct_data['address'] = distinct_data['address'].apply(list)

# # # Display the distinct_data DataFrame
# # print(distinct_data)


463
723


YELP - VEGETARUAN RESTAURANTS 

In [215]:
# Display the distinct values

# Count of unique library names
library_counts = distinct_data[distinct_data['categories'] == 'Libraries']['name'].value_counts()
vegresturant_counts = distinct_data[distinct_data['categories'] != 'Libraries']['name'].value_counts()

print("\nUnique library names count:")
print(library_counts)
print("\nUnique vegetarian resturants names count:")
print(vegresturant_counts)



Unique library names count:
name
Toronto Public Library                                            48
John W. Graham Library                                            23
Ryerson University Student Learning Centre                        23
Gerstein Science Information Centre                               22
Robarts Library                                                   20
Thomas Fisher Rare Book Library                                   19
City Hall Public Library                                          19
EJ Pratt Libary                                                   19
Fort York Library                                                 18
Yorkville Library                                                 14
The Japan Foundation, Toronto                                     14
OISE Library                                                      14
Toronto Reference Library                                         13
Toronto Public Library - Parliament Street Branch                 13


FOURSQUARE - DISPLAY UNIQUE POI NAME, ADDRESS,CATEGORIES 

In [216]:
distinct_data = df_four[['name', 'formatted_address','category']].drop_duplicates()
library_cnt=distinct_data[distinct_data['category']=='Library'].shape[0]
print(library_cnt)
vegrest_cnt=distinct_data[distinct_data['category']!='Library'].shape[0]
print(vegrest_cnt)

140
135


In [217]:
# Display the distinct values
#Count of unique restaurant names
restaurant_counts = distinct_data[distinct_data['category'] != 'Library']['name'].value_counts()

#Count of unique rlibrary names
library_counts = distinct_data[distinct_data['category'] == 'Library']['name'].value_counts()


print("Unique restaurant names count:")
print(restaurant_counts)
print("Unique library names count:")
print(library_counts)

Unique restaurant names count:
name
Freshii                                 17
Kupfert & Kim                            6
Calii Love                               2
Copper Branch                            2
Aesop                                    2
                                        ..
Swiss Chalet Rotisserie & Grill          1
Pho Some Restaurant                      1
Live Food Bar                            1
Ampersand Inc                            1
Flower Lantern Vegetarian Restaurant     1
Name: count, Length: 109, dtype: int64
Unique library names count:
name
Toronto Public Library                41
North York Public Library              4
Corporation of the City of Toronto     3
York Public Library                    3
D Films Corp                           2
                                      ..
Boneshaker Reading Series              1
Function in the Junction               1
Minh Phung                             1
Annette Branch Public Library          1
Woodview 




Foursquare Vegetarina resturants - 140
Foursquare library - 135


Top 10 Vegetarian Friendly Restaurants 
Top 10 Libraries



<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>category</th>
      <th>rating</th>
      <th>reviews</th>
      <th>formatted_address</th>
      <th>telephone</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>9</th>
      <td>Fresh Kitchen + Juice Bar Front</td>
      <td>Vegan and Vegetarian Restaurant</td>
      <td>8.9</td>
      <td>19.0</td>
      <td>47 Front St E (Church Street), Toronto ON M5E 1B3</td>
      <td>(647) 693-7556</td>
    </tr>
    <tr>
      <th>50</th>
      <td>Fresh on Bloor</td>
      <td>Vegan and Vegetarian Restaurant</td>
      <td>8.8</td>
      <td>380.0</td>
      <td>326 Bloor St W, Toronto ON M5S 1W5</td>
      <td>(416) 599-4442</td>
    </tr>
    <tr>
      <th>442</th>
      <td>Tori's Bakeshop</td>
      <td>Bakery</td>
      <td>8.7</td>
      <td>78.0</td>
      <td>2188 Queen St E (Balsam Ave), Toronto ON M4E 1E6</td>
      <td>(647) 350-6500</td>
    </tr>
    <tr>
      <th>10</th>
      <td>Rosalinda</td>
      <td>Mexican</td>
      <td>8.7</td>
      <td>35.0</td>
      <td>133 Richmond St W, Toronto ON M5H 2L3</td>
      <td>(416) 907-0650</td>
    </tr>
    <tr>
      <th>49</th>
      <td>One Love Vegetarian</td>
      <td>Vegan and Vegetarian Restaurant</td>
      <td>8.4</td>
      <td>51.0</td>
      <td>854 Bathurst St (at London St), Toronto ON M5R...</td>
      <td>(416) 535-5683</td>
    </tr>
    <tr>
      <th>478</th>
      <td>Hello 123</td>
      <td>Vegan and Vegetarian Restaurant</td>
      <td>8.4</td>
      <td>36.0</td>
      <td>1122 Queen St W, Toronto ON M6J 1H9</td>
      <td>(416) 532-3555</td>
    </tr>
    <tr>
      <th>345</th>
      <td>YamChops</td>
      <td>Soup Spot</td>
      <td>8.4</td>
      <td>29.0</td>
      <td>705 College St, Toronto ON M6G 1C2</td>
      <td>(416) 645-0117</td>
    </tr>
    <tr>
      <th>403</th>
      <td>Fresh on Crawford</td>
      <td>Vegan and Vegetarian Restaurant</td>
      <td>8.3</td>
      <td>251.0</td>
      <td>894 Queen St W (at Crawford), Toronto ON M6J 1G3</td>
      <td>(416) 599-4442</td>
    </tr>
    <tr>
      <th>255</th>
      <td>Buddha's Vegetarian Foods</td>
      <td>Chinese</td>
      <td>8.3</td>
      <td>48.0</td>
      <td>666 Dundas St W (Bathurst St.), Toronto ON M5T...</td>
      <td>(416) 603-3811</td>
    </tr>
    <tr>
      <th>190</th>
      <td>Fresh on Spadina</td>
      <td>Vegan and Vegetarian Restaurant</td>
      <td>8.2</td>
      <td>419.0</td>
      <td>147 Spadina Ave (at Richmond St. W.), Toronto ...</td>
      <td>(416) 599-4442</td>
    </tr>
  </tbody>
</table>
</div>

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>category</th>
      <th>rating</th>
      <th>reviews</th>
      <th>formatted_address</th>
      <th>telephone</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>59</th>
      <td>Toronto Public Library - Toronto Reference Lib...</td>
      <td>Library</td>
      <td>9.0</td>
      <td>378.0</td>
      <td>789 Yonge St (btwn Collier &amp; Asquith Ave), Tor...</td>
      <td>(416) 395-5577</td>
    </tr>
    <tr>
      <th>189</th>
      <td>Toronto Public Library</td>
      <td>Library</td>
      <td>8.6</td>
      <td>42.0</td>
      <td>190 Fort York Blvd, Toronto ON M5V 0E7</td>
      <td>(416) 393-6240</td>
    </tr>
    <tr>
      <th>30</th>
      <td>Toronto Public Library</td>
      <td>Library</td>
      <td>8.3</td>
      <td>38.0</td>
      <td>239 College St (at Huron St.), Toronto ON M5T 1R5</td>
      <td>(416) 393-7746</td>
    </tr>
    <tr>
      <th>89</th>
      <td>Toronto Public Library</td>
      <td>Library</td>
      <td>8.0</td>
      <td>11.0</td>
      <td>370 Broadview Ave (at Gerrard St E), Toronto O...</td>
      <td>(416) 393-7720</td>
    </tr>
    <tr>
      <th>325</th>
      <td>Toronto Public Library</td>
      <td>Library</td>
      <td>7.9</td>
      <td>51.0</td>
      <td>1101 Bloor St W (at Gladstone), Toronto ON M6H...</td>
      <td>(416) 393-7674</td>
    </tr>
    <tr>
      <th>1597</th>
      <td>Toronto Public Library</td>
      <td>Library</td>
      <td>7.8</td>
      <td>15.0</td>
      <td>3083 Yonge St (at Lawrence Ave.), Toronto ON M...</td>
      <td>(416) 393-7730</td>
    </tr>
    <tr>
      <th>146</th>
      <td>Toronto Public Library</td>
      <td>Library</td>
      <td>7.6</td>
      <td>14.0</td>
      <td>701 Pape Ave (at Danforth Ave), Toronto ON M4K...</td>
      <td>(416) 393-7727</td>
    </tr>
    <tr>
      <th>3687</th>
      <td>Toronto Public Library</td>
      <td>Library</td>
      <td>7.6</td>
      <td>8.0</td>
      <td>496 Birchmount Rd, Toronto ON M1K 1N8</td>
      <td>(416) 396-8890</td>
    </tr>
    <tr>
      <th>443</th>
      <td>Toronto Public Library</td>
      <td>Library</td>
      <td>7.5</td>
      <td>15.0</td>
      <td>2161 Queen St E (at Lee St.), Toronto ON M4L 1J1</td>
      <td>(416) 393-7703</td>
    </tr>
    <tr>
      <th>183</th>
      <td>Toronto Public Library</td>
      <td>Library</td>
      <td>7.5</td>
      <td>12.0</td>
      <td>40 St Clair Ave E (at Yonge St.), Toronto ON M...</td>
      <td>(416) 393-7657</td>
    </tr>
  </tbody>
</table>
</div>


Yelp Vegetarian resturants- 723 entries
Yelp library  - 463
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>categories</th>
      <th>rating</th>
      <th>review</th>
      <th>address</th>
      <th>phone</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>277</th>
      <td>Sunny Cafe</td>
      <td>Vegetarian</td>
      <td>5.0</td>
      <td>3</td>
      <td>40 Saint George Street, Room 6141, Toronto, ON...</td>
      <td></td>
    </tr>
    <tr>
      <th>1908</th>
      <td>KaSpace Cafe</td>
      <td>Vegetarian</td>
      <td>5.0</td>
      <td>9</td>
      <td>765 Queen Street E, Toronto, ON M4M 1H3, Canada</td>
      <td>+16476579600</td>
    </tr>
    <tr>
      <th>3350</th>
      <td>KaSpace Cafe</td>
      <td>Vegetarian</td>
      <td>5.0</td>
      <td>9</td>
      <td>1326 Gerrard Street  E, Toronto, ON M4L 1Z1, C...</td>
      <td>+16476579600</td>
    </tr>
    <tr>
      <th>1931</th>
      <td>Sunny Cafe</td>
      <td>Vegetarian</td>
      <td>5.0</td>
      <td>3</td>
      <td>789 Yonge Street, Toronto, ON M4W 2G8, Canada</td>
      <td></td>
    </tr>
    <tr>
      <th>1</th>
      <td>KaSpace Cafe</td>
      <td>Vegetarian</td>
      <td>5.0</td>
      <td>9</td>
      <td>1432 Gerrard Street East, Toronto, ON M4L 1Z6,...</td>
      <td>+16476579600</td>
    </tr>
    <tr>
      <th>127</th>
      <td>Sunny Cafe</td>
      <td>Vegetarian</td>
      <td>5.0</td>
      <td>3</td>
      <td>2 Bloor Street W, Toronto, ON M4W 3E2, Canada</td>
      <td></td>
    </tr>
    <tr>
      <th>98</th>
      <td>Sunny Cafe</td>
      <td>Vegetarian</td>
      <td>5.0</td>
      <td>3</td>
      <td>10 Spadina Road, Toronto, ON M5R 2S7, Canada</td>
      <td></td>
    </tr>
    <tr>
      <th>198</th>
      <td>KaSpace Cafe</td>
      <td>Vegetarian</td>
      <td>5.0</td>
      <td>9</td>
      <td>Jones 118 Jones Ave, Toronto, ON M4M 2Z9, Canada</td>
      <td>+16476579600</td>
    </tr>
    <tr>
      <th>3031</th>
      <td>Sunny Cafe</td>
      <td>Vegetarian</td>
      <td>5.0</td>
      <td>3</td>
      <td>55 Avenue Rd, Toronto, ON M5R 3L2, Canada</td>
      <td></td>
    </tr>
    <tr>
      <th>971</th>
      <td>Sunny Cafe</td>
      <td>Vegetarian</td>
      <td>5.0</td>
      <td>3</td>
      <td>322 Bloor Street W, Toronto, ON M5S 1W5, Canada</td>
      <td></td>
    </tr>
  </tbody>
</table>
</div>

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>name</th>
      <th>categories</th>
      <th>rating</th>
      <th>review</th>
      <th>address</th>
      <th>phone</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1928</th>
      <td>John W. Graham Library</td>
      <td>Libraries</td>
      <td>5.0</td>
      <td>2</td>
      <td>789 Yonge Street, Toronto, ON M4W 2G8, Canada</td>
      <td>+14169785851</td>
    </tr>
    <tr>
      <th>375</th>
      <td>John M Kelly Library</td>
      <td>Libraries</td>
      <td>5.0</td>
      <td>1</td>
      <td>113 St. Joseph Street, Toronto, ON M5S 3C2, Ca...</td>
      <td></td>
    </tr>
    <tr>
      <th>276</th>
      <td>John W. Graham Library</td>
      <td>Libraries</td>
      <td>5.0</td>
      <td>2</td>
      <td>40 Saint George Street, Room 6141, Toronto, ON...</td>
      <td>+14169785851</td>
    </tr>
    <tr>
      <th>1889</th>
      <td>John W. Graham Library</td>
      <td>Libraries</td>
      <td>5.0</td>
      <td>2</td>
      <td>Toronto Reference Library, 789 Yonge St, 2nd f...</td>
      <td>+14169785851</td>
    </tr>
    <tr>
      <th>65</th>
      <td>John W. Graham Library</td>
      <td>Libraries</td>
      <td>5.0</td>
      <td>2</td>
      <td>10 King's College Road, Room 2402, Toronto, ON...</td>
      <td>+14169785851</td>
    </tr>
    <tr>
      <th>1243</th>
      <td>John M Kelly Library</td>
      <td>Libraries</td>
      <td>5.0</td>
      <td>1</td>
      <td>341 Yonge Street, Toronto, ON M5B 1S1, Canada</td>
      <td></td>
    </tr>
    <tr>
      <th>1249</th>
      <td>John W. Graham Library</td>
      <td>Libraries</td>
      <td>5.0</td>
      <td>2</td>
      <td>341 Yonge Street, Toronto, ON M5B 1S1, Canada</td>
      <td>+14169785851</td>
    </tr>
    <tr>
      <th>1700</th>
      <td>John W. Graham Library</td>
      <td>Libraries</td>
      <td>5.0</td>
      <td>2</td>
      <td>118-92 King Street E, Toronto, ON M5C 2V8, Canada</td>
      <td>+14169785851</td>
    </tr>
    <tr>
      <th>145</th>
      <td>John W. Graham Library</td>
      <td>Libraries</td>
      <td>5.0</td>
      <td>2</td>
      <td>15 King's College Cir, 2nd Fl, N Wing, Toronto...</td>
      <td>+14169785851</td>
    </tr>
    <tr>
      <th>3119</th>
      <td>John W. Graham Library</td>
      <td>Libraries</td>
      <td>5.0</td>
      <td>2</td>
      <td>56 Wellesley St W, Toronto, ON M5S 2S3, Canada</td>
      <td>+14169785851</td>
    </tr>
  </tbody>
</table>
</div>



Yelp is giving more coverage for my location. hence going ahead with yelp.